python/lsst/meas/algorithms/stamps.py

```diff
@@ -0,0 +1,351 @@
+            return cls(stamp_im=stamp_im,
+                       position=SpherePoint(Angle(metadata['RA_DEG'][index], degrees),
+                                            Angle(metadata['DEC_DEG'][index], degrees)),
+                       size=metadata['SIZE'][index])
+        else:
+            return cls(stamp_im=stamp_im, position=SpherePoint(Angle(numpy.nan), Angle(numpy.nan)), size=-1)
+
+
+class StampsBase(abc.ABC, Sequence):
+    """Collection of  stamps and associated metadata.
+
+    Parameters
+    ----------
+    stamps : iterable
+        This should be an iterable of dataclass objects
+        a la ``lsst.meas.algorithms.Stamp``.
+    metadata : `lsst.daf.base.PropertyList`, optional
+        Metadata associated with the bright stars.
+    use_mask : `bool`, optional
+        If ``True`` read and write the mask data.  Default ``True``.
+    use_variance : `bool`, optional
+        If ``True`` read and write the variance data. Default ``True``.
+
+    Notes
+    -----
+    A (gen2) butler can be used to read only a part of the stamps,
+    specified by a bbox:
+
+    >>> starSubregions = butler.get("brightStarStamps_sub", dataId, bbox=bbox)
+    """
+
+    def __init__(self, stamps, metadata=None, use_mask=True, use_variance=True):
+        if not hasattr(stamps, '__iter__'):
+            raise ValueError('The stamps parameter must be iterable.')
+        self._stamps = stamps
+        self._metadata = PropertySet() if metadata is None else metadata.deepCopy()
+        self.use_mask = use_mask
+        self.use_variance = use_variance
+
+    @classmethod
+    @abc.abstractmethod
+    def readFits(cls, filename):
+        """Build an instance of this class from a file.
+
+        Parameters
+        ----------
+        filename : `str`
+            Name of the file to read
+        """
+        raise NotImplementedError
+
+    @classmethod
+    @abc.abstractmethod
+    def readFitsWithOptions(cls, filename, options):
+        """Build an instance of this class with options.
+
+        Parameters
+        ----------
+        filename : `str`
+            Name of the file to read
+        options : `PropertySet`
+            Collection of metadata parameters
+        """
+        raise NotImplementedError
+
+    @abc.abstractmethod
+    def writeFits(self, filename):
+        """Write this object to a file.
+
+        Parameters
+        ----------
+        filename : `str`
+            Name of file to write
+        """
+        raise NotImplementedError
+
+    def __len__(self):
+        return len(self._stamps)
+
+    def __getitem__(self, index):
+        return self._stamps[index]
+
+    def __iter__(self):
+        return iter(self._stamps)
+
+    def append(self, item):
+        """Add an additional bright star stamp.
+
+        Parameters
+        ----------
+        item : `object`
+            Stamp-like object to append.
+        """
+        if not hasattr(item, 'stamp'):
```

Comment:
Would it help if Stamp and BrightStarStamp inherited from AbstractStamp? Then you could do an isinstance check.

Reply:
That's initially how I started, but then the two didn't share any attributes so the only thing that the abstract class would have in it would be an abstract `factory` class method.  I'll do that and you can tell me if you like it or not.